docs/reference/plugins/initrd_plugin.rst

```diff
@@ -0,0 +1,171 @@
+~~~~~~~~~~~~~~~
+
+**Type**: list of strings
+
+A list of firmware to include in the initrd, provided as relative paths to
+``${CRAFT_STAGE}/firmware`` like so:
+
+.. code:: yaml
+
+      initrd-firmware:
+        - foo
+        - bar/baz
+
+
+initrd-modules
+~~~~~~~~~~~~~~
+
+**Type**: list of strings
+
+A list of module to include in the initrd, provided as a list of module names.
+If the specified module(s) have dependencies, they are also installed.
+
+
+Core22 and later options
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+For core24 and later, the following options are also supported. However, they
+will only work if the target architecture has a `systemd-boot-efi <https://packages.ubuntu.com/noble/systemd-boot-efi>`_
+package in the archive. Otherwise, the build will fail.
+
+
+initrd-build-efi-image
+^^^^^^^^^^^^^^^^^^^^^^
+
+**Type**: bool
+
+**Default**: ``false``
+
+If true, a Unified Kernel Image or UKI will be built instead of a standalone
+initrd compressed CPIO archive.
+
+
+initrd-efi-image-key
+^^^^^^^^^^^^^^^^^^^^
+
+**Type**: string
+
+A signing key file to use when creating a UKI EFI image, provided as a relative
+path to ``${CRAFT_STAGE}/signing``.
+
+If set, ``initrd-build-efi-image`` should be ``true`` and
+``initrd-efi-image-cert`` must be specified. If not specified, the snakeoil key
+provided by the ubuntu-core-initramfs package is used.
+
+
+initrd-efi-image-cert
+^^^^^^^^^^^^^^^^^^^^^
+
+**Type**: string
+
+A certificate file to use when creating a UKI EFI image, provided as a relative
+path to ``${CRAFT_STAGE}/signing``.
+
+If set, ``initrd-build-efi-image`` should be ``true`` and
+``initrd-efi-image-key`` must be specified. If not specified, the snakeoil
+certificate provided by the ubuntu-core-initramfs package is used.
+
+
+Environment variables
+---------------------
+
+This plugin does not set its own special variables.
+
+
+Dependencies
+------------
+
+The plugin includes the common build-time requirements for creating a minimal
+initrd, but most of the actual dependencies are installed within the chroot
+environment this plugin builds.
+
+
+How it works
+------------
+
+During the build step the plugin performs the following actions:
+
+#. Pass a collection of flags built from the selected options to an initrd
+   build script within the snapcraft snap.
+#. Fetch a daily compressed archive base of the targeted Ubuntu series.
+#. Unpack the archive into ``${CRAFT_PART_SRC}`` to use as the chroot base.
+#. Copy any kernel firmware or modules from
+   ``${CRAFT_STAGE}/{firmware,modules}`` into the chroot base.
+#. Install any build-time dependencies such as the ubuntu-core-initramfs
+   package into the chroot base.
+#. Any addons, firmware, or modules are added to the chroot base.
+#. ubuntu-core-initramfs is called to create a compressed CPIO archive.
+#. If ``initrd-build-efi-image`` is true, ubuntu-core-initramfs is called again
+   to build the UKI.
+#. The built ``initrd.img`` or UKI is copied to ``${CRAFT_PART_INSTALL}``.
```

Comment:
Minor, but these steps switch to a passive voice.

Reply:
Yes :v